modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/http/conn/LoggingNHttpClientConnection.java

```diff
@@ -105,11 +108,83 @@ public void consumeInput(final NHttpClientEventHandler handler) {
         if (this.log.isDebugEnabled()) {
             this.log.debug(this.id + ": Consume input");
         }
-        super.consumeInput(handler);
-        if (isReleaseConn()) {
-            if (handler instanceof TargetHandler) {
-                ((TargetHandler) handler).getTargetConfiguration().getConnections().releaseConnection(this);
-                this.setReleaseConn(false);
+//        super.consumeInput(handler);
```

Comment:
Why have we removed calling super.consumeInput()? Have we brought that into the logger class? Is it possible to do what we need only here and call super.consumeInput(), reason being, if there are fixes to HTTP Core library at this level, we may have to manually fix those in here. Is there a perf reason for doing this?